elrond-wasm-debug/src/api/managed_types/big_float_api_mock.rs

```diff
@@ -0,0 +1,174 @@
+
+        if f_numerator == None || f_denominator == None {
+            value = f64::from(0);
+        } else {
+            value = f_numerator.unwrap() / f_denominator.unwrap();
+        }
+
+        let mut managed_types = self.m_types_borrow_mut();
+        managed_types.big_float_map.insert_new_handle(value)
+    }
+
+    fn bf_from_sci(&self, significand: i64, exponent: i64) -> Handle {
+        if exponent > 0 {
+            self.signal_error(err_msg::EXPONENT_IS_POSITIVE)
+        }
+
+        let f_significand = significand.to_f64();
+        let value: f64;
+
+        if f_significand == None {
+            value = f64::from(0);
+        } else {
+            let exponent_multiplier = (10.0_f64).powi(exponent as i32);
+            value = f_significand.unwrap() * exponent_multiplier;
+        }
+
+        let mut managed_types = self.m_types_borrow_mut();
+        managed_types.big_float_map.insert_new_handle(value)
+    }
+
+    binary_op_method!(bf_add, add);
+    binary_op_method!(bf_sub, sub);
+    binary_op_method!(bf_mul, mul);
+    binary_op_method!(bf_div, div);
+
+    unary_op_method!(bf_abs, abs);
+    unary_op_method!(bf_neg, neg);
+    fn bf_cmp(&self, x: Handle, y: Handle) -> Ordering {
+        let managed_types = self.m_types_borrow_mut();
+        let bf_x = managed_types.big_float_map.get(x);
+        let bf_y = managed_types.big_float_map.get(y);
+        let order_opt = bf_x.partial_cmp(bf_y);
+        if order_opt == None {
+            self.signal_error(err_msg::CANNOT_COMPARE_VALUES)
+        }
+        order_opt.unwrap()
+    }
+
+    fn bf_sign(&self, x: Handle) -> Sign {
+        let managed_types = self.m_types_borrow();
+        let bf = managed_types.big_float_map.get(x);
+        if !bf.is_normal() {
+            self.signal_error(err_msg::NUMBER_IS_NOT_NORMAL)
+        }
+
+        if bf.is_positive() {
+            return elrond_wasm::api::Sign::Plus;
+        } else if bf.is_negative() {
+            return elrond_wasm::api::Sign::Minus;
+        }
+        elrond_wasm::api::Sign::NoSign
+    }
+
+    fn bf_clone(&self, dest: Handle, x: Handle) {
+        let value = self.bf_get_f64(x);
+        self.bf_overwrite(dest, value);
+    }
+
+    unary_op_method!(bf_sqrt, sqrt);
+
+    fn bf_pow(&self, dest: Handle, x: Handle, exp: i32) {
+        let value = self.bf_get_f64(x);
+        self.bf_overwrite(dest, value.powi(exp));
+    }
+
+    unary_op_method!(bf_floor, floor);
+    unary_op_method!(bf_ceil, ceil);
+    unary_op_method!(bf_trunc, trunc);
+
+    fn bf_is_bi(&self, x: Handle) -> bool {
+        let managed_types = self.m_types_borrow();
+        let bf_x = managed_types.big_float_map.get(x);
+        let trunc_x = bf_x.trunc();
+        let float_trunc_x = trunc_x.to_f64().unwrap();
+        if *bf_x == float_trunc_x {
+            return true;
+        }
+        false
+    }
+
+    fn bf_set_i64(&self, dest: Handle, value: i64) {
+        let f64_value = value.to_f64().unwrap_or_default();
```

Comment:
Since this is a mock and we'd rather know if something went wrong instead of wondering why we get 0, I suggest using unwrap directly.
```suggestion
        let f64_value = value.to_f64().unwrap();
```